common/src/types/wallet/shares.rs

```diff
@@ -29,6 +29,17 @@ impl Wallet {
         circuit_wallet == recovered_wallet
     }
 
+    /// Check that the wallet's blinder and private shares are the result of
+    /// applying a reblind to the old wallet
+    pub fn check_reblind_progression(&self, old_wallet: &Wallet) -> bool {
```

Comment:
i don't think this should be an instance method of the `Wallet` type, the wallet isn't really checking anything. Let's move this to the descriptor, see my second comment